File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.py
def solution(arr, queries):
    answer = []
    
    for i in range(len(queries)):
        answer.append([])
        for j in range(queries[i][0], queries[i][1]+1):
            if arr[j]>queries[i][2]:
                answer[i].append(arr[j])
    result = []
    for i in range(len(answer)):
        if len(answer[i]) == 0:
            result.append(-1)
        else:
            result.append(min(answer[i]))
    return result

File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.java
class Solution {
    public int[] solution(int[] arr, int[][] queries) {
        int[] answer = new int[queries.length];
        for (int i = 0; i < queries.length; i++) {
            int[] query = queries[i];
            int s = query[0];
            int e = query[1];
            int k = query[2];
            int result = -1;
            for (int j = s; j <= e; j++) {
                if (arr[j] > k) {
                    if (result == -1) {
                        result = arr[j];
                    } else {
                        result = Math.min(arr[j], result);
                    }
                }
            }
            answer[i] = result;
        }
        return answer;
    }
}